aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/component/RecipeComponentPreparationService.java

```diff
@@ -169,7 +176,25 @@ public Optional<ComponentOverrideNameVersion> prepare(ComponentOverrideNameVersi
             }
 
             List<Map<String, Object>> manifests = (List<Map<String, Object>>) recipe.get(MANIFESTS);
+
+            // support for multi-platform recipe
```

Comment:
really we only want to do this if the artifacts are different AND an artifact is missing, right?

Reply:
Check for artifact is missing is much simple and easy to implement (from my side). But artifact can be missed also on test JAR build mistake. In that case we just remove corresponding Manifest and probably scenario failed much later (or do not failed at all depends on scenario) which hides original bug.

I can rework to check artifact existence in classpath or file. 
Decision is yours.

Reply:
I'm saying that we only want to _filter_ out a platform if the platform won't work (because the artifact is missing). OTF needs to be able to test multi-platform components which actually have multiple manifests in the recipe (which this change would filter out).

Reply:
So yes, I think the simpler solution is better here. Log a warning if the artifact is missing, but keep the test running.

Reply:
Ok, I will remove current platform detection, platform matching and use successful artifact upload to S3 as key to leave Manifests item. Also I will add warning to log about recipe parts removal to keep user notified.